darshan-runtime/lib/darshan-core.c

```diff
@@ -759,39 +756,1085 @@ void darshan_core_shutdown(int write_log)
         }
 #endif
 
-        darshan_core_fprintf(stderr, "#darshan:<op>\t<nprocs>\t<time>\n");
-        darshan_core_fprintf(stderr, "darshan:log_open\t%d\t%f\n", nprocs, open_tm);
-        darshan_core_fprintf(stderr, "darshan:job_write\t%d\t%f\n", nprocs, job_tm);
-        darshan_core_fprintf(stderr, "darshan:hash_write\t%d\t%f\n", nprocs, rec_tm);
-        darshan_core_fprintf(stderr, "darshan:header_write\t%d\t%f\n", nprocs, header_tm);
-        for(i = 0; i < DARSHAN_MAX_MODS; i++)
+        darshan_core_fprintf(stderr, "#darshan:<op>\t<nprocs>\t<time>\n");
+        darshan_core_fprintf(stderr, "darshan:log_open\t%d\t%f\n", nprocs, open_tm);
+        darshan_core_fprintf(stderr, "darshan:job_write\t%d\t%f\n", nprocs, job_tm);
+        darshan_core_fprintf(stderr, "darshan:hash_write\t%d\t%f\n", nprocs, rec_tm);
+        darshan_core_fprintf(stderr, "darshan:header_write\t%d\t%f\n", nprocs, header_tm);
+        for(i = 0; i < DARSHAN_MAX_MODS; i++)
+        {
+            if(active_mods[i])
+                darshan_core_fprintf(stderr, "darshan:%s_shutdown\t%d\t%f\n",
+                    darshan_module_names[i], nprocs, mod_tm[i]);
+        }
+        darshan_core_fprintf(stderr, "darshan:core_shutdown\t%d\t%f\n", nprocs, all_tm);
+    }
+
+cleanup:
+    for(i = 0; i < DARSHAN_MAX_MODS; i++)
+        if(final_core->mod_array[i])
+            final_core->mod_array[i]->mod_funcs.mod_cleanup_func();
+    darshan_core_cleanup(final_core);
+#ifdef HAVE_MPI
+    if(using_mpi)
+    {
+        free(shared_recs);
+        free(mod_shared_recs);
+    }
+#endif
+    free(logfile_name);
+
+    return;
+}
+
+/* *********************************** */
+
+static void darshan_init_config(struct darshan_core_runtime *core)
```

Comment:
That's actually a good idea, I should have thought of that since you're right that darshan-core source is getting massive. These would be pretty easy to break off into their own source file that operates on Darshan config structures that are populated for use by darshan-core.